src/main/java/com/godaddy/commerce/migrator/handlers/helpers/BatcherHelper.java

```diff
@@ -1,81 +1,81 @@
-
+  public HttpVerbResult<JsonObject> processLegacyEntitlement(final BackfillEntitlementRequest backfillEntitlementRequest, final String customerId, final RoutingContext routingContext) {
+    final String legacyUri = "%s/v2/customers/%s/legacy-fulfillment/processLegacyEntitlement";
+    final HttpVerbWrapperContext wrapperContext = new HttpVerbWrapperContext(
+      LEGACY_FULFILLMENT,
+      routingContext
+    );
+    return wrapperContext.httpVerbWrapper()
+      .header("X-Request-Id", UUID.randomUUID().toString())
+      .header("Authorization", "sso-jwt " + ssojwtHelper.getJWTokenForClientCert(null))
+      .post(JsonObject.class, format(legacyUri, wrapperContext.baseUrl(), customerId), JsonObject.mapFrom(backfillEntitlementRequest));
   }
 
-  public HttpVerbResult<JsonObject> processLegacyEntitlement(
-    JsonObject jsonPayload,
-    String customerId) {
-    String xRequestId = UUID.randomUUID().toString();
-    log.info("ABOUT TO POST TO LEGACY FULFILLMENT SERVICE, X-Request-Id : {}", xRequestId);
-    return HttpVerbWrapper.instance()
-      .setAllTimeouts(Duration.ofSeconds(15))
-      .header("X-Request-Id", xRequestId)
-      .post(JsonObject.class, format(legacyUrl, legacyUrlBase, customerId), jsonPayload);
-  }
+  public String getCustomerIdFromApi(final String shopperId, final RoutingContext routingContext) {
+    final String shopperUri = "/v1/shoppers/%s";
+    final HttpVerbWrapperContext wrapperContext = new HttpVerbWrapperContext(
+      SHOPPERS,
+      routingContext
+    );
 
-  public String getCustomerIdFromApi(String shopperId) {
-    HttpVerbResult<Shopper> response = HttpVerbWrapper.instance()
-      .useSslContext(shopperCert)
+    HttpVerbResult<Shopper> response = wrapperContext.httpVerbWrapper()
       .queryParam("auditClientIp", HOST_ADDRESS)
       .queryParam("includes", "customerId")
-      .get(Shopper.class, format("%s%s", shopperUrl, format(endpoint, shopperId)))
+      .header("Authorization", "sso-jwt " + ssojwtHelper.getJWTokenForClientCert(null))
+      .get(Shopper.class, format("%s%s", wrapperContext.baseUrl(), format(shopperUri, shopperId)))
       .expectsCode(HttpStatus.SC_OK, "Shopper Service call failed");
+
     return response.result().getCustomerId();
   }
 
-  public HttpVerbResult<JsonObject> getEntitlement(String customerId, String entitlementId) {
-    String xRequestId = UUID.randomUUID().toString();
-    return HttpVerbWrapper.instance()
-      .useSslContext(entConn.cert())
-      .header("X-Request-Id", xRequestId)
-      .get(JsonObject.class, format(getEntitlementEndpoint, entitlementsBase, customerId, entitlementId));
+  public HttpVerbResult<JsonObject> getEntitlement(final String customerId, final String entitlementId, RoutingContext routingContext) {
+    final String getEntitlementEndpoint = "%s/v2/customers/%s/entitlements/%s";
+    final HttpVerbWrapperContext wrapperContext = new HttpVerbWrapperContext(
+      ENTITLEMENTS,
+      routingContext
```

Comment:
> [<img alt="mtbennett" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/mtbennett) **Authored by [mtbennett](https://github.secureserver.net/mtbennett)**
_<time datetime="2021-02-03T17:50:49Z" title="Wednesday, February 3rd 2021, 10:50:49 am -07:00">Feb 3, 2021</time>_

---


Ditto on `PREFER_LOCAL_DNS` for a cluster-local target.

```suggestion
      routingContext,
      HttpVerbOption.PREFER_LOCAL_DNS
```